### PR TITLE
Give actual error instead of assert for lang issues. 

### LINKED ISSUE #3525: Need Descriptive Exceptions instead of a assert Trying to use a rule within it's definition.
```
<SocialMagnet>
    SocialMagnet
    ...
```
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/24756432-need-descriptive-exceptions-instead-of-a-assert-trying-to-use-a-rule-within-it-s-definition?utm_campaign=plugin&utm_content=tracker%2F42681&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F42681&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by matham**
Fixes #3525.

### CODE REVIEW

File: kivy/lang.py [hunk 1]
```diff
@@ -1932,13 +1932,20 @@ def _apply_rule(self, widget, rule, rootrule, template_ctx=None):
         # rootrule: the current root rule (for children of a rule)
 
         # will collect reference to all the id in children
-        assert(rule not in self.rulectx)
+        if rule in self.rulectx:
+            raise ParserException(
+                rule.ctx, rule.line,
+                'Class cannot be a child of itself'.format(rule.name))
```
Comment (dessant) [inline kivy/lang.py]: A placeholder is missing: `Class {} cannot be`.

File: kivy/lang.py [hunk 2]
```diff
@@ -1932,13 +1932,20 @@ def _apply_rule(self, widget, rule, rootrule, template_ctx=None):
...
             'ids': {'root': widget.proxy_ref},
             'set': [], 'hdl': []}
 
         # extract the context of the rootrule (not rule!)
-        assert(rootrule in self.rulectx)
+        if rootrule in self.rulectx:
+            raise ParserException(
+                rootrule.ctx, rootrule.line,
+                'Class cannot be a child of itself'.format(rootrule.name))
+        print rootrule
```
Comment (dessant) [inline kivy/lang.py]: leftover

**Comment by matham** [comment]
I dunno, this doesn't like right to me.

**Comment by dessant** [comment]
@matham, please make up your mind, there is one week left until closure. :rose:

**Comment by matham** [comment]
Meh.

**Pull request closed without merging**